enhancements/20220822-storage-network/20220822-storage-network.md

```diff
@@ -0,0 +1,134 @@
+# Storage Network
+
+## Summary
+
+Longhorn storage will need sync data to every replica engine. The sync traffic will be heavy and occupy too many bandwidth from management interface. To keep management interface being reliable. Storage Network will help to separate traffic to another interfaces.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/1055
+https://github.com/longhorn/longhorn/issues/2285
+
+## Motivation
+
+### Goals
+
+- Longhorn will sync replica from storage network
+
+### Non-goals [optional]
+
+- Set storage network in the installation stage
+
+## Proposal
+
+### User Stories
+
+Assume Harvester nodes will have 4 network interfaces (eth0, eth1, eth2, eth3).
+Users want to have eth0 and eth1 to be management interface with HA.
+And eth2, eth3 are for VM and Storage Network.
+
+VM and Storage Network will have different VLAN ID.
+
+#### Story 1
+
+VM and Storage Network will use the same interfaces but different VLAN IDs.
+
+- VM and Storage want to share eth2 and eth3.
+- Create VlanConfig with eth2 and eth3
+- Create Network-Attatchment-Definition for VM
+- Create Network-Attatchment-Definition with Whereabouts config for Storage
+
+#### Story 2
+
+VM and Storage Network will use different interfaces.
+
+- VM use eth2 and eth3
+- Storage uses eth4 and eth5
+- Create a new VlanConfig with eth4 and eth5
+- Create Network-Attatchment-Definition with Whereabouts config for Storage only
+
+### User Experience In Detail
+
+#### Manual
+
+1. Users have to shutdown all VMs and stop all Pods attached with Longhorn volumes manually before setting Longhorn Storage Network. (Including Grafana and Prometheus)
+2. Users need to assign a bridge from VLAN Config and a VLAN ID for the storage network.
+3. Setting Longhorn setting for storage network.
+4. Restart pods and VMs manually.
+
+#### Controller
+
+1. Users have to shotdown all VMs and stop all VMI before setting `storage-network` in Harvester settings
+2. Users need to assign a bridge from VLAN Config and a VLAN ID for storage network (Need to setup IPAM Whereabouts IP range)
+3. Controller will shutdown management pods attached with Longhorn volumes and check all volumes are detached.
+4. Controller will set `storage-network` config to Longhorn settings `storage-network`.
+5. Controller will restart management pods.
+6. Users need to restart VM manually.
+
+### API changes
+
+- New Settings `storage-network`
```

Comment:
The revert of setting seems not described cleary.
Assue the `storage-network` is ready, user wishes to change it back to default mode, namely, no `stoarge-network`.
For most of `Harvester settgings`, they support `set to default value`.
How can user remove existing `storage-network` setting/set back default.

Reply:
revert config is the same as setting it.

Reply:
HEP needs to add story for it, wirte down the detailed steps.

Well, HEP is to define black-box view of input/output, user will have clear understanding of the operation steps and outpus.

Reply:
fix